### PR TITLE
Fourth reconciliation PR from production/RRFS.v1

### PR DESCRIPTION
**Pull request by grantfirl**
## Description

This work corresponds to https://github.com/NOAA-EMC/fv3atm/pull/802 merged into production/RRFS.v1, except this is targeted at the develop branch. This was @haiqinli 's work.

-- Update the MYNN PBL, Smoke and subgrid cloud fraction for the RRFS.v1 code freeze.
-- New (not in RRFSv1): update ccpp-framework commit hash (only updates required cmake version)

What bug does it fix, or what feature does it add?
-- Merge the bugfix in the HAILCAST diagnostic code (unit issue) (https://github.com/NOAA-EMC/fv3atm/issues/320)
-- Merge (https://github.com/NOAA-EMC/fv3atm/pull/801)

Is a change of answers expected from this PR? Yes.  

### Issue(s) addressed

Non

## Testing

Original testing for production/RRFS.v1 merge: 

How were these changes tested? The changes were tested with retro runs and regression test.

What compilers / HPCs was it tested with? Intel compiler on Hera and Jet.

Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Yes.

Have the ufs-weather-model regression test been run? On what platform? Yes, on Hera.

Will the code updates change regression test baseline? If yes, why? Please show the baseline directory below.
Yes, because the physics changed.
Please commit the regression test log files in your ufs-weather-model branch
Yes, the regression test log file was committed.

New testing for develop branch merge:
Ran full rt.conf on Hera. The test_changes.list and RT log were pushed into the UFSWM PR.

## Dependencies

https://github.com/ufs-community/ccpp-physics/pull/230

# Requirements before merging
- [ ] All new code in this PR is tested by at least one unit test
- [ ] All new code in this PR includes Doxygen documentation
- [ ] All new code in this PR does not add new compilation warnings (check CI output)

### CODE REVIEW

File: ccpp/data/GFS_typedefs.meta [hunk 1]
```diff
@@ -2367,23 +2367,23 @@
   standard_name = kinematic_surface_upward_sensible_heat_flux_of_fire
   long_name = kinematic surface upward sensible heat flux of fire
   units = K m s-1
-  dimensions = (horizontal_loop_extent)
```
Comment (grantfirl) [inline ccpp/data/GFS_typedefs.meta:2372]: @grantfirl Note to self: This was messed up in https://github.com/NOAA-EMC/fv3atm/pull/815

Comment (grantfirl) [reply ccpp/data/GFS_typedefs.meta:2372]: This fix causes the fire behavior RT to change results. This restores the correct functionality for the two-way fire coupling.

**Comment by DusanJovic-NOAA** [comment]
@grantfirl Can you please update ccpp-framework hash in one of your 'reconciliation' PRs to point to the top of the main branch. There has been only one commit that updates minimum cmake version to eliminate come cmake warnings. See https://github.com/NCAR/ccpp-framework/pull/611

**Comment by grantfirl** [comment]
> @grantfirl Can you please update ccpp-framework hash in one of your 'reconciliation' PRs to point to the top of the main branch. There has been only one commit that updates minimum cmake version to eliminate come cmake warnings. See [NCAR/ccpp-framework#611](https://github.com/NCAR/ccpp-framework/pull/611)

Of course. I'll do it here.

File: ccpp/data/GFS_typedefs.F90 [hunk 2]
```diff
@@ -2371,7 +2377,7 @@ subroutine sfcprop_create (Sfcprop, Model)
     allocate (Sfcprop%hprime   (IM,Model%nmtvr))
     allocate (Sfcprop%dust12m_in  (IM,12,5))
     allocate (Sfcprop%smoke_RRFS(IM,24,2))
-    allocate (Sfcprop%smoke2d_RRFS(IM,4))
+    allocate (Sfcprop%smoke2d_RRFS(IM,5))
     allocate (Sfcprop%emi_in   (IM,1))
     allocate(Sfcprop%albdirvis_lnd (IM))
```
Comment (BrianCurtis-NOAA) [inline ccpp/data/GFS_typedefs.F90]: spacing inconcistent here between `allocate` and `(`

**Review by BrianCurtis-NOAA — APPROVED** [review]

File: ccpp/data/GFS_typedefs.F90 (hunk 2 above)
Comment (grantfirl) [reply ccpp/data/GFS_typedefs.F90]: I made all allocate statements consistent in both GFS_ and CCPP_typedefs.F90 (e.g. `allocate (...)`)

Comment (BrianCurtis-NOAA) [reply ccpp/data/GFS_typedefs.F90]: Much appreciated!

**Review by jkbk2004 — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Pull request merged**